tests/extras/a-torus.test.js

```diff
@@ -0,0 +1,44 @@
+/* global assert, suite, test, setup, process */
+suite('a-torus', function () {
+  suite('Basic funcions', function () {
+    var torus;
+    setup(function (done) {
+      var scene = document.createElement('a-scene');
+      torus = document.createElement('a-torus');
+      document.body.appendChild(scene);
+      scene.appendChild(torus);
+      done();
+    });
+    test('Torus has default position when created', function (done) {
+      process.nextTick(function () {
+        assert.deepEqual(torus.getComputedAttribute('position'), { x: 0, y: 0, z: 0 });
+        done();
+      });
+    });
+    test('Torus element has torus geometry', function (done) {
+      process.nextTick(function () {
+        assert.strictEqual(torus.attributes.geometry.value, 'primitive:torus');
+        done();
+      });
+    });
+  });
+});
+/* global assert, suite, test, setup, process */
+suite('a-torus: custom attributes', function () {
+  var torus;
+  setup(function (done) {
+    var scene = document.createElement('a-scene');
+    torus = document.createElement('a-torus');
+    torus.setAttribute('segments-tubular', '100');
+    torus.setAttribute('radius', '2');
+    document.body.appendChild(scene);
+    scene.appendChild(torus);
+    done();
+  });
+  test('Torus geometry inherits attributes', function (done) {
+    process.nextTick(function () {
+      assert.strictEqual(torus.attributes.geometry.value, 'primitive:torus;segmentsTubular:100;radius:2');
```

Comment:
I would use getAttribute to test so it's not a hard-coded string dependent on order